### ISSUE TITLE
📎 Port `no-negation-in-equality-check` from `eslint-plugin-unicorn`

### ISSUE THREAD

**Issue by esaari** [biomejs/biome#10897]
### Description

Please consider porting [`unicorn/no-negation-in-equality-check`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-negation-in-equality-check.md).

It catches a real precedence bug rather than a stylistic preference: a negation on the left operand of an equality check binds tighter than the comparison, so `!foo === bar` evaluates as `(!foo) === bar` — almost always a mistake for `foo !== bar`.

**Invalid**
```js
if (!foo === bar) {}
if (!foo !== bar) {}
```

**Valid**
```js
if (foo !== bar) {}
if (!(foo === bar)) {}
```

- Operators: `===` and `!==` (strict equality).
- Because two corrections are plausible (`!foo === bar` → `foo !== bar`, or wrap as `!(foo === bar)`), upstream offers them as **suggestions** rather than a single autofix — so this would be suggestion-only in Biome too.

Suggested name/category: `noNegationInEqualityCheck` under **suspicious** (it flags likely bugs, not just style) — happy to defer to your naming/category conventions.

If it's in scope, I'm willing to implement it (I'm also picking up #3939 / `useConsistentErrorSubclassing`).

### Source
- `eslint-plugin-unicorn` / `no-negation-in-equality-check`

**Comment by dyc3** [comment]
Yeah this would be a great rule to have. Should be pretty simple too. Feel free to send a PR!

**Comment by esaari** [comment]
Thanks @dyc3 — opened #10907 with the rule (detection-only, strict `===`/`!==` with a negated left operand, matching the source rule's scope).

**Comment by PhoenixMputu** [comment]
I have implemented this rule and opened a Pull Request for it here: #10919

It includes the unsafe rule logic with both requested code actions (flipping the operator or adding parentheses), along with full snapshot test coverage. Looking forward to your feedback!

**Comment by Functionhx** [comment]
For visibility, there are currently two open implementations for this issue: #10915 and #10919.

I’m the author of #10915. It was opened earlier and has already received an approval from @dyc3. Since that approval, I made several non-trivial follow-up changes to strengthen the autofix implementation, including additional ASI safety handling, improved comment/trivia preservation, handling hazards introduced by generated parentheses, and expanded regression coverage.

The latest revision is `d9fc280a`, and I’ve requested a re-review because the fix logic changed after the original approval.

I’m posting this status update mainly to avoid duplicated review effort. Thanks to @PhoenixMputu as well for exploring an alternative implementation. I’ll leave the final implementation choice to the maintainers.
